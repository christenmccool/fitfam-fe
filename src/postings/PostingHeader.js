import React, {useState} from 'react';

import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import { styled } from '@mui/material/styles';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';


//From MUI docs
const ExpandMore = styled((props) => {
  const { expand, ...other } = props;
  return <IconButton {...other} />;
})(({ theme, expand }) => ({
  transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
  marginLeft: 'auto',
  transition: theme.transitions.create('transform', {
    duration: theme.transitions.duration.shortest,
  }),
}));

/** Shows information about a posting 
 * - Posting date
 * - Workout name and description
 * 
 * Used by PostingDetail and ResultDetail
 * 
 * PostingDetail -> PostingHeader -> ResultCardList -> ResultCard
 * ResultDetail -> PostingHeader -> ResultInfo -> {CommentCardList -> CommentCard, CommentForm} 
 *
 * Routed at /postings/:id
 */
const PostingHeader = ({ postDate, woName, woDescription }) => {
  const [expanded, setExpanded] = useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Box>
      <Typography variant="h6" color="text.secondary" mb={1}>
        {postDate}
      </Typography>
      <Typography variant="h3" color="secondary" >
        {woName}
      </Typography>
      <Box sx={{display: 'flex', justifyContent: 'flex-end'}}>
        <ExpandMore
          expand={expanded}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </ExpandMore>
      </Box>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <Typography variant="h6" mb={3} style={{whiteSpace: "pre-wrap", wordWrap: "break-word"}}>
          {woDescription}
        </Typography>
      </Collapse>
    </Box>
  )
}

export default PostingHeader;
