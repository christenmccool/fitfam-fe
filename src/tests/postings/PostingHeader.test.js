import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import PostingHeader from "../../postings/PostingHeader";

// const PostingHeader = ({ postDate, woName, woDescription, famName="" }) => {

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <PostingHeader 
        postDate="Tuesday, March 22nd, 2022"
        woName="Test Workout Name 1"
        woDescription="Do this hard thing"
        famName="Family 1"
      />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <PostingHeader 
        postDate="Tuesday, March 22nd, 2022"
        woName="Test Workout Name 1"
        woDescription="Do this hard thing"
        famName="Family 1"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders posting details", async function () {
  render(
    <MemoryRouter>
      <PostingHeader 
        postDate="Tuesday, March 22nd, 2022"
        woName="Test Workout Name 1"
        woDescription="Do this hard thing"
        famName="Family 1"
      />
    </MemoryRouter>
  );
  expect(screen.queryByText("Tuesday, March 22nd, 2022")).toBeInTheDocument();
  expect(screen.queryByText("Family 1")).toBeInTheDocument();
  expect(screen.queryByText("Test Workout Name 1")).toBeInTheDocument();
});







