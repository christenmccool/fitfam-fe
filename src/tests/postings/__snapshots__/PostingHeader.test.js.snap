// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1krercm-MuiTypography-root"
    >
      Tuesday, March 22nd, 2022
    </h6>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-wulekh-MuiTypography-root"
    >
      Family 1
    </h4>
    <div
      class="MuiBox-root css-0"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1t5mc0a-MuiTypography-root"
      >
        Test Workout Name 1
      </h3>
      <div
        class="MuiBox-root css-1bvc4cc"
      >
        <button
          aria-expanded="false"
          aria-label="show more"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-3ogxn6-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
